### PR TITLE
EP-2213 Affiliate AMP Product Card

### PR DESCRIPTION
**Pull request by Mwindo**
Support for AMP version of Affiliate product card. (Update to https://github.com/orgnc/wordpress-plugin/pull/68)
Screen Shot 2022-06-22 at 7 10 26 PM

### CODE REVIEW

File: Organic/Organic.php [hunk 1]
```diff
@@ -1269,6 +1280,17 @@ public function syncAdsTxt() {
...
+    public function syncAffiliateConfig() {
+        $config = $this->sdk->queryAffiliateConfig();
+
+        $this->debug( 'Got affiliate domain: ' . $config['publicDomain'] );
+        $this->updateOption( 'organic::affiliate_domain', $config['publicDomain'], false );
+
+        return array(
+            'updated' => true,
+        );
+    }
```
Comment (Mwindo) [inline Organic/Organic.php:1292]: We can sync other fields as needed

File: Organic/Organic.php [hunk 2]
```diff
@@ -1269,6 +1281,30 @@ public function syncAdsTxt() {
...
+            self::captureException( $e );
+            return array(
+                'updated' => false,
+            );
+        } catch ( \Exception $e ) {
+            self::captureException( $e );
+            return array(
+                'updated' => false,
+            );
+        }
```
[the remark this answers is not in the document]
Comment (Mwindo) [reply Organic/Organic.php:1297]: Got it. I wasn't sure how we handled multiple exceptions, and IDE made this suggestion.

File: Organic/AmpAffiliateInjector.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                frameborder="0"
+                sandbox="allow-scripts allow-same-origin allow-popups allow-popups-to-escape-sandbox"
+                src="$url"
+            >
+                <p placeholder="">Loading iframe content</p>
+            </amp-iframe>
+HTML;
+        $fragment = $this->dom->createDocumentFragment();
+        $fragment->appendXML( $amp_iframe_code );
+        $product_card_div->appendChild( $fragment );
```
[the remark this answers is not in the document]
Comment (Mwindo) [reply Organic/AmpAffiliateInjector.php:73]: Thanks, @sergunich!

File: Organic/SDK/OrganicSdk.php [hunk 4]
```diff
@@ -10,6 +10,8 @@
 use GraphQL\Variable;
 use GraphQL\Query;
 use GraphQL\RawObject;
+use GuzzleHttp\Client as RestClient; // let's switch to GraphQL in the future
```
[the remark this answers is not in the document]
Comment (Mwindo) [reply Organic/SDK/OrganicSdk.php:13]: I'll keep it as GET for now and update if/when we switch to GraphQL.

**Review by sergunich — APPROVED** [review]
👍

**Pull request merged**